x-pack/plugins/observability_solution/inventory/public/hooks/use_detail_view_redirect.ts

```diff
@@ -45,59 +43,58 @@ export const useDetailViewRedirect = () => {
   const serviceOverviewLocator = locators.get<ServiceOverviewParams>(SERVICE_OVERVIEW_LOCATOR_ID);
 
   const getSingleIdentityFieldValue = useCallback(
-    (entity: Entity) => {
-      const identityFields = castArray(entity[ENTITY_IDENTITY_FIELDS]);
+    (latestEntity: InventoryEntityLatest) => {
+      const identityFields = castArray(latestEntity.entityIdentityFields);
       if (identityFields.length > 1) {
-        throw new Error(`Multiple identity fields are not supported for ${entity[ENTITY_TYPE]}`);
+        throw new Error(
+          `Multiple identity fields are not supported for ${latestEntity.entityType}`
+        );
       }
 
       const identityField = identityFields[0];
-      return entityManager.entityClient.getIdentityFieldsValue(unflattenEntity(entity))[
+      return entityManager.entityClient.getIdentityFieldsValue(toEntityLatest(latestEntity))[
```

Comment:
I'm not happy about mapping `InventoryEntityLatest` into `EntityLatest`, but I'm not sure if I want to change the interface the the entity client works with.